spec/hardware/main.fmf

```diff
@@ -0,0 +1,91 @@
+story:
+    As a tester I want to specify detailed hardware requirements
+    for the guest on which the tests will be executed.
+
+description: |
+    As part of the :ref:`/spec/plans/provision` step it is
+    possible to use the ``hardware`` key to specify additional
+    requirements for the testing environment. This provides a
+    generic and an extensible way to write down essential hardware
+    requirements. For example one consistent way how to specify
+    `at least 2 GB of RAM` for all supported provisioners.
+
+    Individual requirements are provided as a simple ``key:
+    value`` pairs, for example the minimum amount of
+    :ref:`/spec/hardware/memory`, or the related information is
+    grouped under a common parent, for example ``cores`` or
+    ``model`` under the :ref:`/spec/hardware/cpu` key.
+
+    **Search**
+
+    Regular expressions can be used for selected fields such as
+    the ``model-name``. Please note, that the full extent of
+    regular expressions might not be supported across all
+    provision implementations. The ``.*`` notation, however,
+    should be supported everywhere::
+
+        # Enable regular expression search
+        cpu:
+            model-name: =~ .*AMD.*
```

Comment:
https://github.com/teemtee/tmt/issues/909

Reply:
Thanks for reminding this. Added my comment there.

Reply:
The regular expression part updated in c802af0c.